tests/clv/models/test_beta_geo.py

```diff
@@ -160,63 +118,133 @@ def test_expected_probability_alive(self):
         The "true" prefix refers to the value obtained using 1) the closed form
         solution and 2) the data-generating parameter values.
         """
-        true_prob_alive = self.alive.mean()  # scalar
-        est_prob_alive = self.fixed_model.expected_probability_alive(
-            self.customer_id,
-            self.frequency,
-            self.recency,
-            self.T,
+        rng = np.random.default_rng(152)
+
+        N = 100
+        # Almost deterministic p = .02
+        a = 0.02 * 10_000
+        b = 0.98 * 10_000
+        alpha = 3
+        r = 4
+
+        recency, frequency, alive, T = self.generate_data(a, b, alpha, r, N, rng=rng)
+        customer_id = list(range(N))
+
+        bg_model = BetaGeoModel(
+            customer_id=customer_id,
+            frequency=frequency,
+            recency=recency,
+            T=T,
+            a_prior=pm.Normal.dist(a, 1e-3),
+            b_prior=pm.Normal.dist(b, 1e-3),
+            alpha_prior=pm.Normal.dist(alpha, 1e-3),
+            r_prior=pm.Normal.dist(r, 1e-3),
         )
 
-        assert est_prob_alive.shape == (1, 1000, self.N)
+        fake_fit = pm.sample_prior_predictive(
+            samples=500, model=bg_model.model, random_seed=rng
+        )
+        fake_fit.add_groups(dict(posterior=fake_fit.prior))
+        bg_model._fit_result = fake_fit
+
+        est_prob_alive = bg_model.expected_probability_alive(
+            customer_id,
+            frequency,
+            recency,
+            T,
+        )
+
+        assert est_prob_alive.shape == (1, 500, N)
         assert est_prob_alive.dims == ("chain", "draw", "customer_id")
 
         np.testing.assert_allclose(
-            true_prob_alive.mean(),
+            alive.mean(),
             est_prob_alive.mean(),
             rtol=0.05,
         )
 
     def test_expected_num_purchases(self):
```

Comment:
What's the justification for combining these two tests? These are two separate methods with different input and output dimensions.

Reply:
They both require building the same type of values, patching the pymc-makerting and lifetimes model, and they are both tested against the same lifetimes method. There was a TODO before suggesting they be merged already, probably for similar reasoning

Reply:
Done